internal/extension/registry/extension_manager.go

```diff
@@ -0,0 +1,211 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package registry
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	corev1 "k8s.io/api/core/v1"
+	k8smachinery "k8s.io/apimachinery/pkg/types"
+	k8scli "sigs.k8s.io/controller-runtime/pkg/client"
+	k8sclicfg "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/envoygateway"
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	extTypes "github.com/envoyproxy/gateway/internal/extension/types"
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
+	"github.com/envoyproxy/gateway/proto/extension"
+)
+
+const grpcServiceConfig = `{
+"methodConfig": [{
+	"name": [{"service": "envoygateway.extension.EnvoyGatewayExtension"}],
+	"waitForReady": true,
+	"retryPolicy": {
+		"MaxAttempts": 4,
+		"InitialBackoff": "0.1s",
+		"MaxBackoff": "1s",
+		"BackoffMultiplier": 2.0,
+		"RetryableStatusCodes": [ "UNAVAILABLE" ]
+	}
+}]}`
+
+var _ extTypes.Manager = (*Manager)(nil)
+
+type Manager struct {
+	k8sClient     k8scli.Client
+	namespace     string
+	extension     v1alpha1.Extension
+	hookConnCache map[extTypes.ExtensionXDSHookType]*grpc.ClientConn
```

Comment:
shouldnt this a xdsHookConnCache ?
how would pre & post hook state be stored ?
would a separate cache be created for each hook type ?

Reply:
Yeah I think you're right and we don't need a separate cache for each hook since there is only a single extension and the connection info is the same regardless of what hook is being called.